### DISCUSSION TITLE
Saving model parameters

### DISCUSSION THREAD

**Discussion by pedronarloch** [autogluon/autogluon#3234] [Q&A] [answered]
Hello there, I'm currently trying out AutoGluon with some time series data and would like to know if it is possible to get the model parameters after running the whole predictor.fit(). Why? I would like to have the flexibility of instantiating an arbitrary model that AutoGluon uses without relying on the Predictor/Learner/Trainer. Is it possible?

I'm posting a small code snippet on how I'm currently organizing the model search script. Notice that I would save the best model parameters in the last block of code. This way, I would not need to use the .pkl files and could train/re-train the model with other data.

```python
from autogluon.timeseries.predictor import TimeSeriesPredictor

# Load your time series data into a DataFrame, where the first column represents the timestamp
data = ...

# Specify the name of the timestamp column in your DataFrame
time_column = 'timestamp'

# Create the AutoGluon TimeSeriesPredictor
predictor = TimeSeriesPredictor(
    label='target',
    forecast_length=10,  # specify the length of the forecast horizon
    time_column=time_column,
    eval_metric='rmse',
    ...  # other settings and hyperparameters
)

# Train the AutoGluon model
predictor.fit(train_data=data)

# Saving model parameters
...
```

Thank you very much for helping, and keep pushing toward this nice AutoML project.

**Comment by shchur** [comment, marked as answer]
Hello, unfortunately, this is currently not supported as a part of the public API. You could try the following workaround to extract the model with the best validation score
```python
...
predictor.fit(data, enable_ensemble=False)

trainer = predictor._trainer
best_model = trainer.load_model(trainer.model_best)
```
This will give you a model object that you could try using outside of AutoGluon. However, many things that you described (e.g. retraining) might be difficult. AutoGluon relies on many models coming from different libraries, and we put significant work into wrapping them with a unified API. Many of the models do not support incremental training, and often can't be summarized just by its parameters (e.g., `ETS/ARIMA/Theta` are always refit from scratch for every new time series).

Note that we cannot provide guarantees that this will work properly, as it's not a part of the public API.

As a side note, the code snippet you provided would need to be modified to run correctly. E.g., replacing `forecast_length` with `prediction_length`, removing the `time_column` argument, and changing `eval_metric="RMSE"`.

**Comment by pedronarloch** [comment]
Thank you for the quick answer @shchur!
